game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProResourceTracker.java

```diff
@@ -64,7 +64,7 @@ public boolean isEmpty() {
   }
 
   public int getTempPUs(final GameState data) {
-    final Resource pus = data.getResourceList().getResource(Constants.PUS);
+    final Resource pus = data.getResourceList().getResource(Constants.PUS).orElse(null);
```

Comment:
@BBlaza Same, either the method assumes the value must be available, then new method `getResourceOrThrow?` or a default value should be used, e.g., `orElse(0)`.